### PR TITLE
Bash completion for flags

### PR DESCRIPTION
**Pull request by mabelzhang**
# 🎉 New feature

Part of https://github.com/gazebosim/gz-tools/issues/1
Used together with https://github.com/gazebosim/gz-tools/pull/87

## Summary

Bash completion, installation, tests.

## Test it

```
$ . install/share/gz/gz.completion
$ ign msg -     
--force-version  --info           --versions       -i
--help           --list           -h               -l
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/cmd/msgs.bash_completion.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
```
Comment (ahcorde) [inline src/cmd/msgs.bash_completion.sh:1]: LICENSE

File: tools/ign_TEST.cc [hunk 2]
```diff
@@ -89,6 +90,30 @@ TEST(CmdLine, MsgInfo)
...
+  auto output = custom_exec_str("ign msg --help --force-version " + g_version);
+  EXPECT_NE(std::string::npos, output.find("--info")) << output;
+  EXPECT_NE(std::string::npos, output.find("--list")) << output;
+  EXPECT_NE(std::string::npos, output.find("--help")) << output;
+  EXPECT_NE(std::string::npos, output.find("--force-version")) << output;
+  EXPECT_NE(std::string::npos, output.find("--versions")) << output;
+
+  // Flags in bash completion
+  std::ifstream scriptFile(std::string(PROJECT_SOURCE_PATH) +
+    "/src/cmd/msgs.bash_completion.sh");
```
Comment (ahcorde) [inline tools/ign_TEST.cc:106]: `ignition::common::JoinPaths`

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: tools/ign_TEST.cc (hunk 2 above)
Comment (mabelzhang) [reply tools/ign_TEST.cc:106]: This package also does not have dependency on ign-common. There's a comment like this in another source file:

https://github.com/gazebosim/gz-msgs/blob/7e1b8d463ad00430e046f997bcce64315838ba9b/src/Utility.cc#L37-L39

I don't know if we should do a `QUIET` find like in gz-plugin or hardcode the `/` like tests there.

File: src/cmd/msgs.bash_completion.sh (hunk 1 above)
Comment (mabelzhang) [reply src/cmd/msgs.bash_completion.sh:1]: e3cb342

File: tools/ign_TEST.cc (hunk 2 above)
Comment (mabelzhang) [reply tools/ign_TEST.cc]: 744a6a2 uses `std::filesystem::path`

**Review by ahcorde — APPROVED** [review]

**Review by chapulina — APPROVED** [review]
LGTM! Pushed 385141ed6bc10b1f880c6e49f047a4a9be4509dc with some final tweaks:

* Always install CLI files, see https://github.com/gazebosim/gz-msgs/pull/166 and https://github.com/gazebo-tooling/release-tools/issues/472
* Link against filesystem to fix Bionic build

**Comment by scpeters** [comment]
I think the debian metadata needs to be updated to install the new completion files

* [Build Status](https://build.osrfoundation.org/job/ign-msgs5-debbuilder/174/) https://build.osrfoundation.org/job/ign-msgs5-debbuilder/174/console

~~~
dh_missing: warning: usr/share/gz/gz1.completion.d/msgs5.bash_completion.sh exists in debian/tmp but is not installed to anywhere
~~~

**Pull request merged**